airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeDestination.java

```diff
@@ -25,90 +25,32 @@
 package io.airbyte.integrations.destination.snowflake;
 
 import com.fasterxml.jackson.databind.JsonNode;
-import com.google.common.base.Charsets;
-import com.google.common.collect.Lists;
 import io.airbyte.commons.json.Jsons;
-import io.airbyte.commons.lang.CloseableQueue;
-import io.airbyte.commons.resources.MoreResources;
 import io.airbyte.db.jdbc.JdbcDatabase;
 import io.airbyte.integrations.base.Destination;
-import io.airbyte.integrations.base.DestinationConsumer;
 import io.airbyte.integrations.base.IntegrationRunner;
-import io.airbyte.integrations.destination.DestinationConsumerFactory;
-import io.airbyte.integrations.destination.NamingConventionTransformer;
-import io.airbyte.integrations.destination.SqlDestinationOperations;
-import io.airbyte.protocol.models.AirbyteConnectionStatus;
-import io.airbyte.protocol.models.AirbyteConnectionStatus.Status;
-import io.airbyte.protocol.models.AirbyteMessage;
-import io.airbyte.protocol.models.AirbyteRecordMessage;
-import io.airbyte.protocol.models.ConfiguredAirbyteCatalog;
-import io.airbyte.protocol.models.ConnectorSpecification;
-import java.io.IOException;
-import java.sql.PreparedStatement;
-import java.sql.SQLException;
-import java.sql.Timestamp;
-import java.time.Instant;
-import java.util.List;
-import java.util.UUID;
+import io.airbyte.integrations.destination.jdbc.AbstractJdbcDestination;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class SnowflakeDestination implements Destination {
+public class SnowflakeDestination extends AbstractJdbcDestination implements Destination {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(SnowflakeDestination.class);
 
-  protected static final String COLUMN_NAME = "data";
-
-  private final SnowflakeSQLNameTransformer namingResolver;
-
   public SnowflakeDestination() {
-    namingResolver = new SnowflakeSQLNameTransformer();
-  }
-
-  @Override
-  public ConnectorSpecification spec() throws IOException {
-    final String resourceString = MoreResources.readResource("spec.json");
-    return Jsons.deserialize(resourceString, ConnectorSpecification.class);
+    // the driver class is a no op because we override getDatabase.
+    super("", new SnowflakeSQLNameTransformer(), SnowflakeSqlOperations::new);
   }
 
   @Override
-  public AirbyteConnectionStatus check(JsonNode config) {
-    try {
-      final JdbcDatabase database = SnowflakeDatabase.getDatabase(config);
-      database.execute("SELECT 1;");
-      return new AirbyteConnectionStatus().withStatus(Status.SUCCEEDED);
-    } catch (Exception e) {
-      return new AirbyteConnectionStatus().withStatus(Status.FAILED).withMessage(e.getMessage());
-    }
+  protected JdbcDatabase getDatabase(JsonNode config) {
+    return SnowflakeDatabase.getDatabase(config);
   }
 
+  // this is a no op since we override getDatabase.
```

Comment:
this isn't quite as pretty as i'd like it to be, but since we're just stubbing in no ops, i'm not inclined to get too crazy about it.